pkg/apis/k0s/v1beta1/images_test.go

```diff
@@ -106,28 +106,77 @@ func TestImagesRepoOverrideInConfiguration(t *testing.T) {
 }
 
 func TestOverrideFunction(t *testing.T) {
-	repository := "my.registry"
-	testCases := []struct {
-		Input  string
-		Output string
-	}{
-		{
-			Input:  "repo/image",
-			Output: "my.registry/repo/image",
-		},
-		{
-			Input:  "registry.com/repo/image",
-			Output: "my.registry/repo/image",
-		},
-		{
-			Input:  "image",
-			Output: "my.registry/image",
-		},
-	}
+	t.Run("overrideRepository without path", func(t *testing.T) {
+		repository := "my.registry"
+		testCases := []struct {
+			Input  string
+			Output string
+		}{
+			{
+				Input:  "repo/image",
+				Output: "my.registry/repo/image",
+			},
+			{
+				Input:  "registry.com/repo/image",
+				Output: "my.registry/repo/image",
+			},
+			{
+				Input:  "image",
+				Output: "my.registry/image",
+			},
+		}
+
+		for _, tc := range testCases {
+			assert.Equal(t, tc.Output, overrideRepository(repository, tc.Input))
+		}
+	})
+	t.Run("overrideRepository with path", func(t *testing.T) {
+		repository := "my.registry/foo"
+		testCases := []struct {
+			Input  string
+			Output string
+		}{
+			{
+				Input:  "repo/image",
+				Output: "my.registry/foo/repo/image",
+			},
+			{
+				Input:  "registry.com/repo/image",
+				Output: "my.registry/foo/repo/image",
+			},
+			{
+				Input:  "image",
+				Output: "my.registry/foo/image",
+			},
+		}
+		for _, tc := range testCases {
+			assert.Equal(t, tc.Output, overrideRepository(repository, tc.Input))
+		}
+	})
+	t.Run("overrideRepository with repo path and double invocation", func(t *testing.T) {
+		repository := "my.registry/foo"
+		testCases := []struct {
+			Input  string
+			Output string
+		}{
+			{
+				Input:  "repo/image",
+				Output: "my.registry/foo/repo/image",
+			},
+			{
+				Input:  "registry.com/repo/image",
+				Output: "my.registry/foo/repo/image",
+			},
+			{
+				Input:  "image",
+				Output: "my.registry/foo/image",
+			},
+		}
+		for _, tc := range testCases {
+			assert.Equal(t, tc.Output, overrideRepository(repository, overrideRepository(repository, tc.Input)))
```

Comment:
Maybe call the function twice? The test data does test idempotency, but I think testing twice would be better

Reply:
> Maybe call the function twice?
Isn't this calling it twice already? Or you mean something like:
```
for _, tc := range testCases {
			img := overrideRepository(repository, tc.Input)
			assert.Equal(t, tc.Output, img)
			img2 := overrideRepository(repository, img)
			assert.Equal(t, tc.Output, img2)
		}
```

Reply:
I just realized the test casees for the previous test are exactly the same so it's effectively doing the same thing. Disregard this.